### PR TITLE
Tooling refresh and CI improvements

### PR DESCRIPTION
**Pull request by cen1**
- reproducible build test
- cross platform build tests
- conan v1 -> v2 update
- readme improvements
- release CI workflow (deb, rpm) on tag
- minor compile warn operator precedent fix, some more code analysis and warn fixes after this mr
- bump cmake requirement

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,35 +1,62 @@
...
+    "src/bncsutil/file.h"
+    "src/bncsutil/keytables.h"
+    "src/bncsutil/libinfo.h"
+    "src/bncsutil/ms_stdint.h"
+    "src/bncsutil/mutil.h"
+    "src/bncsutil/mutil_types.h"
+    "src/bncsutil/nls.h"
+    "src/bncsutil/oldauth.h"
+    "src/bncsutil/pe.h"
+    "src/bncsutil/sha1.c"
```
Comment (carlbennett) [inline CMakeLists.txt]: The files `sha1.c` and `sha1.h` seem to be specified but only `sha1.h` was supposed to be, it looks like.

Perhaps `sha1.c` should be specified under `SOURCES`?

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,35 +1,62 @@
...
+    "src/bncsutil/cdkeydecoder.cpp"
+    "src/bncsutil/checkrevision.cpp"
+    "src/bncsutil/decodekey.cpp"
+    "src/bncsutil/file.cpp"
+    "src/bncsutil/libinfo.cpp"
+    "src/bncsutil/oldauth.cpp"
 )
 
-add_library(${PROJECT_NAME} ${SOURCE} ${HEADERS})
+target_sources(bncsutil PRIVATE ${SOURCES} ${HEADERS})
```
Comment (carlbennett) [inline CMakeLists.txt:44]: **Inconsistency:** Seemed to have removed `${PROJECT_NAME}` here but not in other places like line 82.

Should the lines [44, 53, 59, 70, 71, 75, 76, 79] replace `bncsutil` with `${PROJECT_NAME}`?

File: src/bncsutil/mutil.h [hunk 3]
```diff
@@ -51,7 +51,7 @@
 #define    ROR(x,n)    _lrotr((x),(n))
 #else
 #ifndef ROL
-#define ROL(a,b) (((a) << (b)) | ((a) >> 32 - (b)))
+#define ROL(a,b) (((a) << (b)) | ((a) >> (32 - (b))))
 #endif
 #ifndef ROR
 #define ROR(a,b) (((a) >> (b)) | ((a) << 32 - (b)))
```
Comment (carlbennett) [inline src/bncsutil/mutil.h:57]: If `ROL` is updated with parentheses, shouldn't `ROR` be updated with parentheses?

**Review by carlbennett — COMMENTED** [review]
Thanks for your submission, here are some suggestions that could improve this change.

File: src/bncsutil/mutil.h (hunk 3 above)
Comment (cen1) [reply src/bncsutil/mutil.h:57]: will do more of the warn removals in a later commit, this one was just generating the most warns

File: CMakeLists.txt (hunk 2 above)
Comment (cen1) [reply CMakeLists.txt:44]: `${PROJECT_NAME}` was a "smart" idea of mine in the past but I've never seen this in modern cmake in other projects so better use static. Fixed in https://github.com/BNETDocs/bncsutil/pull/18/commits/9fd0649220e8d1fc55e356dab8af5c8deb89841c

File: CMakeLists.txt (hunk 1 above)
Comment (cen1) [reply CMakeLists.txt]: correct, fixed in https://github.com/BNETDocs/bncsutil/pull/18/commits/bc7cc2a78414fa2840e34baec458e17b215d9142

**Review by carlbennett — APPROVED** [review]

**Pull request merged**